src/views/design/ScriptDetail/index.tsx

```diff
@@ -498,22 +509,31 @@ const ScriptDetail = withStyles(styles)(
                         <GenericDraggableList
                             listItems={listItems}
                             columns={columns}
-                            listActions={[
-                                {
-                                    icon: <EditIcon />,
-                                    label: translator('scripts.detail.main.list.item.actions.edit'),
-                                    onClick: (id, index) => {
-                                        this.setState({ editActionIndex: index });
-                                    },
-                                },
-                                {
-                                    icon: <DeleteIcon />,
-                                    label: translator('scripts.detail.main.list.item.actions.delete'),
-                                    onClick: (id, index) => {
-                                        this.setState({ actionIndexToDelete: index });
+                            listActions={[].concat(
+                                !checkAuthority(SECURITY_PRIVILEGES.S_SCRIPTS_WRITE, 'PUBLIC')
+                                && checkAuthority(SECURITY_PRIVILEGES.S_SCRIPTS_READ, 'PUBLIC')
+                                    ? {
+                                        icon: <Visibility />,
+                                        label: translator('scripts.detail.main.list.item.actions.edit'),
+                                        onClick: (index: number) => {
+                                            this.setState({ editActionIndex: index });
+                                        },
+                                    } : {
+                                        icon: <EditIcon />,
+                                        label: translator('scripts.detail.main.list.item.actions.edit'),
+                                        onClick: (index: number) => {
+                                            this.setState({ editActionIndex: index });
+                                        },
                                     },
-                                },
-                            ]}
+                                checkAuthority(SECURITY_PRIVILEGES.S_SCRIPTS_WRITE, 'PUBLIC')
+                                    ? {
+                                        icon: <DeleteIcon />,
+                                        label: translator('scripts.detail.main.list.item.actions.delete'),
+                                        onClick: (index: number) => {
+                                            this.setState({ actionIndexToDelete: index });
+                                        },
+                                    } : [],
+                            )}
```

Comment:
Can we already use the security group name  of the script in question instead of the fixed value 'PUBLIC'.
Also, refactor the logic
```
if (checkAuthority(SECURITY_PRIVILEGES.S_SCRIPTS_WRITE, 'PUBLIC')) {
    logic
} else {
    if (checkAuthority(SECURITY_PRIVILEGES.S_SCRIPTS_READ, 'PUBLIC')) {
         logic
    } else {
         null
    }
}
```